test/async.test.mo

```diff
@@ -36,69 +43,84 @@ func state(handler : TokenHandler.TokenHandler) : (Nat, Nat, Nat) {
   );
 };
 
+func createHandler(triggerOnNotifications : Bool) : (TokenHandler.TokenHandler, Journal) {
+  let journal : Vec.Vector<(Time.Time, Principal, TokenHandler.LogEvent)> = Vec.new();
+  TokenHandler.TokenHandler(
+    ledger,
+    anon_p,
+    0,
+    triggerOnNotifications,
+    func(logInfo : (Principal, TokenHandler.LogEvent)) {
+      Vec.add(journal, (Time.now(), logInfo.0, logInfo.1));
+    },
+  ) |> (_, journal);
+};
+
 module Debug {
   public func state(handler : TokenHandler.TokenHandler) {
     print(
       debug_show handler.state()
     );
   };
-  public func journal(handler : TokenHandler.TokenHandler, ctr : Nat) {
+  public func journal(journal : Journal, startFrom : Nat) {
     print(
       debug_show (
-        handler.queryJournal(?ctr)
+        Vec.toArray(journal)
+        |> Array.slice(_, startFrom, _.size())
+        |> Iter.toArray(_)
       )
     );
   };
 };
 
 do {
-  let handler = TokenHandler.TokenHandler(ledger, anon_p, 1000, 0, false);
+  let (handler, journal) = createHandler(false);
```

Comment:
Maybe we can even create `inc` inside `createHandler`? Because the two lines
```
let (handler, journal) = createHandler(false);
let (inc, _) = create_inc();
```
always appear together.

Or we could make `Journal` a class, have the inc-related counter inside, and provide a function for assertions about the length. Then also the lines 
```
assert Vec.size(journal) == 0;
```
will become even shorter. For example they could look like
```
assert journal.hasSize(5);
```